File: typescript/src/graphqlClient.ts
import { performance } from "perf_hooks";
import { gql, request } from "graphql-request";

const baseURL = "http://127.0.0.1:5002/graphql";
const testCounts = [100, 200, 300];

const queries = {
  usuarios: gql`
    {
      usuarios {
        ID
        nome
      }
    }
  `,
  musicas: gql`
    {
      musicas {
        ID
        nome
      }
    }
  `,
  playlists: gql`
    {
      playlists {
        ID
        nome
      }
    }
  `,
};

async function testLoad(query: any, name: string, numTest: number) {
  const start = performance.now();

  for (let i = 0; i < numTest; i++) {
    await request(baseURL, query).catch((error) =>
      console.error(`Error: ${error.message}`)
    );
  }

  const end = performance.now();
  console.log(
    `${name} load test with ${numTest} requests completed in ${
      (end - start) / 1000
    } seconds`
  );
}

export async function runTests() {
  for (let numTest of testCounts) {
    console.log(`Running tests with ${numTest} requests:`);
    await testLoad(queries.usuarios, "Usuarios", numTest);
    await testLoad(queries.musicas, "Musicas", numTest);
    await testLoad(queries.playlists, "Playlists", numTest);
  }
}

File: csharp/GraphqlClient.cs
using System;
using System.Threading.Tasks;
using GraphQL.Client.Http;
using GraphQL.Client.Serializer.Newtonsoft;

public class GraphqlClient
{
    private readonly GraphQLHttpClient _client;

    public GraphqlClient(string baseAddress)
    {
        _client = new GraphQLHttpClient(
            new GraphQLHttpClientOptions { EndPoint = new Uri(baseAddress) },
            new NewtonsoftJsonSerializer()
        );
    }

    private async Task SendRequestAsync(GraphQLHttpRequest request, int i)
    {
        try
        {
            var response = await _client.SendQueryAsync<dynamic>(request);
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Request {i} failed: {ex.Message}");
        }
    }

    public async Task RunTests()
    {
        var queries = new[]
        {
            new { Name = "Usuarios", Query = "{ usuarios { ID nome } }" },
            new { Name = "Musicas", Query = "{ musicas { ID nome } }" },
            new { Name = "Playlists", Query = "{ playlists { ID nome } }" }
        };

        var numTests = new[] { 100, 200, 300 };

        foreach (var numTest in numTests)
        {
            Console.WriteLine($"{numTest} requests:");
            foreach (var query in queries)
            {
                var request = new GraphQLHttpRequest(query.Query);
                var start = DateTime.Now;

                for (int i = 0; i < numTest; i++)
                {
                    await SendRequestAsync(request, i);
                }

                var end = DateTime.Now;
                Console.WriteLine($"{query.Name} : {(end - start).TotalSeconds} seconds");
            }
        }
    }
}
